### PR TITLE
Support http-client-0.5

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

**Comment by kRITZCREEK** [comment]
Won't this cause any connection failure to crash the `psc-ide-server`? I checked the release notes and it looks like the exception code has changed, but exceptions still need to be caught here.

**Comment by phadej** [comment]
Ah, true. One want to catch all exceptions... I'll revert that functionality

**Comment by phadej** [comment]
reverted

**Comment by phadej** [comment]
I have no idea why osx tests failed,  could they be restarted?

**Comment by kRITZCREEK** [comment]
Yeah that's not a genuine failure. There's something flaky compiling psc-ide's test assets...

**Comment by kRITZCREEK** [comment]
I hope the tests will work reliably again after #2286.

File: src/Language/PureScript/Ide/Pursuit.hs [hunk 1]
```diff
@@ -35,41 +35,37 @@ import qualified Pipes.Prelude                 as P
...
-  let req = req'
-        { queryString= "q=" <> (fromString . T.unpack) qClean
-        , requestHeaders=[(hAccept, "application/json")]
-        }
-  m <- newManager tlsManagerSettings
-  withHTTP req m $ \resp ->
-    P.fold (<>) "" identity (responseBody resp)
-
+    let qClean = T.dropWhileEnd (== '.') q
+    req' <- parseRequest "http://pursuit.purescript.org/search"
```
Comment (paf31) [inline src/Language/PureScript/Ide/Pursuit.hs]: Can this be changed to `https` now?

**Comment by paf31** [comment]
LGTM, other than my https comment. @kRITZCREEK good to merge then?

**Comment by kRITZCREEK** [comment]
Looks good from my side :+1:

File: src/Language/PureScript/Ide/Pursuit.hs (hunk 1 above)
Comment (phadej) [inline src/Language/PureScript/Ide/Pursuit.hs]: I'll leave this for someone else to change.

**Comment by paf31** [comment]
Thanks @phadej!

**Pull request merged**